graylog2-web-interface/src/components/events/events/EventDetails.tsx

```diff
@@ -74,6 +94,15 @@ const EventDetails = ({ event, eventDefinitionContext }: Props) => {
                   Replay search
                 </ReplaySearchButton>
               </dd>
+              <AddEvidence id={event.id}
+                           type="events"
+                           investigationSelected={(val: boolean) => setDisableAddEvidence(!val)}>
```

Comment:
Instead of calling `investigationSelected` and maintaining a state, we could just provide this information for the children:
`<AddEvidence>{({ investigationSelected }) => <dd>}</AddEvidence>`

For cases where we do not need this information, we could still support the current structure, by checking if `children` is a function in `AddEvidence`.
`typeof children === 'function' ? children({ investigationSelected }) : children`

What do you think?

Reply:
I really like this pattern. I have seen it, but never implemented it. It makes more sense and it is cleaner. I'll make this change. Thanks! Excellent suggestion.